MdeModulePkg/Core/Dxe/Misc/DebugImageInfo.c

```diff
@@ -269,33 +268,33 @@ CoreRemoveDebugImageInfoEntry (
   EFI_HANDLE  ImageHandle
   )
 {
-  EFI_DEBUG_IMAGE_INFO        *Table;
-  UINTN                       Index;
-  EFI_DEBUG_IMAGE_INFO_NORMAL *NormalImage;
+  EFI_DEBUG_IMAGE_INFO          *Table;
+  UINTN                         Index;
+  EFI_DEBUG_IMAGE_INFO_NORMAL2  *NormalImage2;
 
   mDebugInfoTableHeader.UpdateStatus |= EFI_DEBUG_IMAGE_INFO_UPDATE_IN_PROGRESS;
 
   Table = mDebugInfoTableHeader.EfiDebugImageInfoTable;
 
   for (Index = 0; Index < mMaxTableEntries; Index++) {
-    if ((Table[Index].NormalImage != NULL) && (Table[Index].NormalImage->ImageHandle == ImageHandle)) {
+    if ((Table[Index].NormalImage2 != NULL) && (Table[Index].NormalImage2->ImageHandle == ImageHandle)) {

```

Comment:
This will work for different types (e.g. 1 & 2) in the table, if all types have ImageHandle at the same offset. Not that there are ever going to be different types in the table, I don't think - but not sure if some thing should be done to document this?

Reply:
Actually, probably better to resolve this in code, if only with an assert.